# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu_bindings/src/server.rs
# Commit: 41d7ed93df76
# Full Hash: 41d7ed93df76d84a8e842758f852d291a908e065
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2023-10-17 03:58:35
# Regressor Bug: 1838693
# File Overlap Count: 2
# Description:
#   Bug 1838693 Part 3: Make GPUDevice::Destroy() trigger wgpu device_destroy, then device_drop. r=webgpu-reviewers,nical
#   
#   This treats destroy as a 2-step process: wait on the destroy, then drop.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190238
# ==============================================================================

diff -r f214290956c1 -r 41d7ed93df76 gfx/wgpu_bindings/src/server.rs
--- a/gfx/wgpu_bindings/src/server.rs	Mon Oct 16 15:32:00 2023 +0000
+++ b/gfx/wgpu_bindings/src/server.rs	Mon Oct 16 15:32:00 2023 +0000
@@ -227,6 +227,11 @@
 }
 
 #[no_mangle]
+pub extern "C" fn wgpu_server_device_destroy(global: &Global, self_id: id::DeviceId) {
+    gfx_select!(self_id => global.device_destroy(self_id))
+}
+
+#[no_mangle]
 pub extern "C" fn wgpu_server_device_drop(global: &Global, self_id: id::DeviceId) {
     gfx_select!(self_id => global.device_drop(self_id))
 }
